frontend/simple/utils/distribution/mincome-default.js

```diff
@@ -49,20 +50,27 @@ function transferValue (incomes, indexAmount, transferAmount) {
   })
 }
 
-function incomeKeys (a, b) {
-  return a.name > b.name
+function incomeKeys (a: Object, b: Object): number {
+  if (a.name > b.name) {
+    return 1
+  }
+  if (a.name < b.name) {
+    return -1
+  }
+  return 0
 }
 
 function incomeAmounts (a, b) {
   return a.amount - b.amount
 }
 
-function floorTo (number) {
-  return number.toString().match(/^-?\d+(?:\.\d{0,2})?/)[0]
+function floorTo (number): number {
+  const numberMatch = number.toString().match(/^-?\d+(?:\.\d{0,2})?/)
+  return numberMatch ? parseFloat(numberMatch[0]) : 0;
 }
 
-function incomeDistribution (incomes, minCome) {
-  let incomeLength = Object.keys(incomes).length
+function incomeDistribution (incomes: Array<Object>, minCome: number) {
```

Comment:
Can we validate the keys inside each object? Something like...

```
incomes: Array<{ amount: number }>
```

Not sure if this is the correct syntax

Reply:
yes. done. thanks for the tip!